### PR TITLE
Reorganize plans constants

### PR DESCRIPTION
**Pull request by sgomes**
This PR reorganizes the data in `lib/plans/constants`, creating new files (`features-list.js` and `plans-list.js`) and changing where the exported functions live.

No data should have been changed; only the files where it lives.

These changes allow us to remove the feature data in plans (not all plans data, unfortunately) from the `build` chunk, removing it from e.g. `/log-in`. It could also enable more optimizations in the future, as the dependencies within `lib/plans` become clearer.

#### Changes proposed in this Pull Request

* Move feature data to new `lib/plans/features-list.js`
* Move plan data to new `lib/plans/plans-list.js`
* Make `lib/plans/constants` not depend on anything
* Move exports around to where they make the most sense after the above
* Stop `lib/plans/index.js` depending on `lib/plans/features-list.js`
* Rewrite imports across the codebase to match all of the above
* Remove use of plan feature data for `propTypes` validation

#### Testing instructions

* Test as much plan functionality as possible throughout the application. There should be no errors or visible changes.

### CODE REVIEW

**Comment by sgomes** [comment]
ICFY seems to indicate this would result in ~4KB less compressed JS for endpoints like `/log-in`: http://iscalypsofastyet.com/branch?branch=update/reorganize-plans-constants

**Comment by sgomes** [comment]
Rebased after the changes in #31802

**Comment by jsnajdr** [comment]
I investigated why we're still getting tiny changes like:
```
google-my-business           +1 B   (+0.0%)       +2 B   (+0.0%)
```
and the culprit is export name mangling.
```js
import { GROUP_JETPACK } from 'lib/plans/constants';
```
is tranformed either to:
```js
lib_plans_constants__WEBPACK_IMPORTED_MODULE_3__[/* GROUP_JETPACK */ "mb"]
```
or to:
```js
lib_plans_constants__WEBPACK_IMPORTED_MODULE_3__[/* GROUP_JETPACK */ "Ac"]
```
The `mb` and `Ac` are numeric indexes of the exports converted to strings.

Webpack 5 will make this a lot better: the mangling can be turned off (`optimization.mangleExports` option) and when turned on, the mangled names will be deterministic and stable, i.e., hashes of the export name rather that of its numeric index.

**Comment by sgomes** [comment]
E2E tests seem to be failing due to reasons unrelated to this PR. They're being fixed in #32277.

**Review by blowery — APPROVED** [review]
LGTM. Would love someone else to validate it though.

**Comment by yoavf** [comment]
Not a full review, but I did successfully test a monthly WPCOM Business plan purchase (+switch to yearly) just to make sure things in that less-frequented part are still working :)

**Comment by sgomes** [comment]
Adding `DO NOT MERGE` tag temporarily as I rebase on top of continued changes to `constants.js` 🙁

**Comment by sgomes** [comment]
Removing the `DO NOT MERGE` tag, as the updated code appears to test well.

**Comment by sgomes** [comment]
> Not a full review, but I did successfully test a monthly WPCOM Business plan purchase (+switch to yearly) just to make sure things in that less-frequented part are still working :)

Thank you for taking a look, @yoavf ! 🙂

File: client/lib/plans/index.js
Comment (jsnajdr) [inline client/lib/plans/index.js]: Nice attempt to get rid of `lodash.get` 🙂 I'm wondering if `getPathSlug` needs to be a method at all: just a simple `pathSlug` property would be sufficient. The same applies for the `getStoreSlug` and `getProductId` methods.

**Review by jsnajdr — APPROVED** [review]
Nice PR that should be merged ASAP to prevent further conflicts.

One idea for further improvement: the `PLANS_LIST` export is redundant. At many places it's used just for `PropTypes` and we can remove it there. At other places the `getPlans` exported function can be used.

File: client/lib/plans/index.js
Comment (sgomes) [reply client/lib/plans/index.js]: Ah, this method changed files at some point during my PR, so I guess it doesn't make a lot of sense anymore to drop the `lodash.get` dependency, given that this file makes so much usage of Lodash 😁But this at least avoids string parsing, so I'll keep it 🙂 

As for `getPathSlug` and friends, I'm guessing they're kept as methods to facilitate A/B testing? I opted against changing the data structure itself, since the plans list and individual plans are being depended on directly, and it's hard to ensure there isn't any code depending on the fact that this property is a method.

**Comment by sgomes** [comment]
> One idea for further improvement: the `PLANS_LIST` export is redundant. At many places it's used just for `PropTypes` and we can remove it there. At other places the `getPlans` exported function can be used.

Yes, that export was already marked as something that shouldn't be imported directly, but there are a number of places making use of it :-/ We still won't be able to get rid of it entirely, though, even after this PR, unless we merge `lib/plans/index.js` and `lib/plans/plans-list.js`

File: client/lib/plans/index.js
Comment (jsnajdr) [reply client/lib/plans/index.js]: Grepping the sources for `getPathSlug`, I don't see any place where the method call would provide any extra value over a `pathSlug` property. It always returns a constant and is never overridden or customized.

It's quite common that a piece of code used to make sense some time ago but it doesn't today. Refactorings and other changes happen and not all cleanup opportunities are spotted.

Comment (sgomes) [reply client/lib/plans/index.js]: Properties are a good point. If the value needed to be determined at runtime, there would be no need for a function; a property with a getter would be enough 👍

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/2040487).

**Pull request merged**
